pkg/reconciler/apis/apiexportendpointslice/apiexportendpointslice_controller.go

```diff
@@ -57,13 +57,16 @@ const (
 // NewController returns a new controller for APIExportEndpointSlices.
 // Shards and APIExports are read from the cache server.
 func NewController(
-	apiExportEndpointSliceClusterInformer apisinformers.APIExportEndpointSliceClusterInformer,
+	shardName string,
+	apiExportEndpointSliceClusterInformer apisv1alpha1informers.APIExportEndpointSliceClusterInformer,
+	apiBindingInformer apisv1alpha1informers.APIBindingClusterInformer,
```

Comment:
is this needed in the non-url controller?

Reply:
It is. When binding changed, it might be sign of new consumer or one going away

Reply:
no its not.